bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/metastore/BigQueryMetastoreCatalog.java

```diff
@@ -0,0 +1,328 @@
+  private FileIO fileIO;
+  private Object conf;
+  private String projectId;
+  private String location;
+  private String defaultWarehouseLocation;
+  private BigQueryClient client;
+  private boolean filterUnsupportedTables;
+
+  // Must have a no-arg constructor to be dynamically loaded
+  // initialize(String name, Map<String, String> properties) will be called to complete
+  // initialization
+  public BigQueryMetastoreCatalog() {}
+
+  @Override
+  public void initialize(String inputName, Map<String, String> properties) {
+    if (!properties.containsKey(GCPProperties.PROJECT_ID)) {
+      throw new ValidationException("GCP project must be specified");
+    }
+
+    this.projectId = properties.get(GCPProperties.PROJECT_ID);
+    this.location = properties.getOrDefault(GCPProperties.BIGQUERY_LOCATION, DEFAULT_GCP_LOCATION);
+    BigQueryClient bigQueryClient;
+    try {
+      bigQueryClient = new BigQueryClientImpl();
+    } catch (IOException e) {
+      throw new ServiceFailureException(e, "Creating BigQuery client failed");
+    } catch (GeneralSecurityException e) {
+      throw new ValidationException(e, "Creating BigQuery client failed due to a security issue");
+    }
+
+    initialize(inputName, properties, projectId, location, bigQueryClient);
+  }
+
+  @VisibleForTesting
+  void initialize(
+      String inputName,
+      Map<String, String> properties,
+      String gcpProjectId,
+      String gcpLocation,
+      BigQueryClient bigQueryClient) {
+    this.catalogPluginName = inputName;
+    this.catalogProperties = ImmutableMap.copyOf(properties);
+    this.projectId = gcpProjectId;
+    this.location = gcpLocation;
+    this.client =
+        Preconditions.checkNotNull(bigQueryClient, "Failed to initialize BigQuery Client");
+
+    LOG.info("Using BigQuery Metastore Iceberg Catalog: {}", inputName);
+
+    if (properties.containsKey(CatalogProperties.WAREHOUSE_LOCATION)) {
+      // Iceberg always removes trailing slash to avoid paths like "<folder>//data" in file systems
+      // like s3.
+      this.defaultWarehouseLocation =
+          LocationUtil.stripTrailingSlash(properties.get(CatalogProperties.WAREHOUSE_LOCATION));
+    }
+
+    String fileIoImpl =
+        properties.getOrDefault(
+            CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.hadoop.HadoopFileIO");
+    this.fileIO = CatalogUtil.loadFileIO(fileIoImpl, properties, conf);
+
+    this.filterUnsupportedTables =
+        Boolean.parseBoolean(
+            properties.getOrDefault(PROPERTIES_KEY_FILTER_UNSUPPORTED_TABLES, "false"));
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    return new BigQueryTableOperations(
+        client,
+        fileIO,
+        projectId,
+        // Sometimes extensions have the namespace contain the table name too, so we are forced to
+        // allow invalid namespace and just take the first part here like other catalog
+        // implementations do.
```

Comment:
Can you point to other cases of this? It seems incorrect to me. If the namespace contains the table name after being passed in here, it is not correct.